backend/pkg/api/routes.go

```diff
@@ -175,3 +206,27 @@ func (api *API) routes() *chi.Mux {
 
 	return baseRouter
 }
+
+// NewRequestValidationInterceptor creates an interceptor to validate Connect requests.
+func NewRequestValidationInterceptor(_ *zap.Logger, validator *protovalidate.Validator) connect_go.UnaryInterceptorFunc {
+	interceptor := func(next connect_go.UnaryFunc) connect_go.UnaryFunc {
+		return connect_go.UnaryFunc(func(
+			ctx context.Context,
+			req connect_go.AnyRequest,
+		) (connect_go.AnyResponse, error) {
+			msg, ok := req.Any().(protoreflect.ProtoMessage)
+			if !ok {
+				return nil, connect_go.NewError(connect_go.CodeInvalidArgument, errors.New("request is not a protocol buffer message"))
```

Comment:
In theory if everything is configured properly and compiles, it should not be possible realistically... but without the check the process would panic otherwise since it's a runtime error.